app/src/main/java/com/android/sample/ui/createRecipe/CategoryScreen.kt

```diff
@@ -0,0 +1,171 @@
+
+          // Title
+          Text(
+              text = stringResource(R.string.select_category),
+              style = Typography.displayLarge,
+              color = MaterialTheme.colorScheme.onPrimary,
+              modifier =
+                  Modifier.fillMaxWidth()
+                      .padding(horizontal = PADDING_16.dp)
+                      .testTag("CategoryTitle"),
+              textAlign = TextAlign.Center)
+
+          Spacer(modifier = Modifier.weight(1f))
+
+          // Subtitle
+          Text(
+              text = stringResource(R.string.select_category_description_optional),
+              style = MaterialTheme.typography.bodyMedium,
+              color = MaterialTheme.colorScheme.onPrimary,
+              modifier = Modifier.padding(horizontal = PADDING_32.dp).testTag("CategorySubtitle"),
+              textAlign = TextAlign.Center)
+
+          Spacer(modifier = Modifier.weight(2f))
+
+          // Dropdown menu for selecting category
+          Box(modifier = Modifier.fillMaxWidth().padding(horizontal = PADDING_16.dp)) {
+            OutlinedButton(
+                onClick = { expanded.value = !expanded.value },
+                modifier = Modifier.fillMaxWidth().testTag(DROPDOWN_TEST_TAG),
+                shape = RoundedCornerShape(DROPDOWN_CORNER_RADIUS.dp),
+                colors =
+                    ButtonDefaults.outlinedButtonColors(
+                        containerColor = MaterialTheme.colorScheme.surface,
+                        contentColor = MaterialTheme.colorScheme.onSurface)) {
+                  Text(
+                      text =
+                          selectedCategory.value ?: stringResource(R.string.no_category_selected),
+                      style = MaterialTheme.typography.bodyMedium,
+                      modifier = Modifier.padding(vertical = PADDING_8.dp))
+                }
+
+            DropdownMenu(
+                expanded = expanded.value,
+                onDismissRequest = { expanded.value = false },
+                modifier =
+                    Modifier.fillMaxWidth(0.9f)
+                        .fillMaxHeight(0.3f)
+                        .background(MaterialTheme.colorScheme.surface)) {
+                  categories.forEachIndexed { index, category ->
+                    DropdownMenuItem(
+                        text = {
+                          Text(text = category, style = MaterialTheme.typography.bodyMedium)
+                        },
+                        onClick = {
+                          selectedCategory.value =
+                              if (category == noCategoryString) null else category
+                          expanded.value = false
+                        },
+                        modifier = Modifier.testTag("DropdownMenuItem_$category"))
+                    if (index < categories.size - 1) {
+                      Divider(color = MaterialTheme.colorScheme.onSurface.copy(alpha = 0.12f))
```

Comment:
The Divider is deprecated, you can replace it with an `HorizontalDivider`, don't forget to define the `alpha? in the Dimension object within the constants file (resources/C.kt).:
```suggestion
                      HorizontalDivider(color = MaterialTheme.colorScheme.onSurface.copy(alpha = 0.12f))
                      
```